.github/workflows/shellcheck.yaml

```diff
@@ -0,0 +1,28 @@
+name: shellcheck
+on:
+  push:
+    branches:
+      - main 
+  pull_request:
+permissions: {}
+
+jobs:
+  shellcheck:
+    name: shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: shellcheck workflow-scripts
+        uses: ludeeus/action-shellcheck@master
+        env:
+          SHELLCHECK_OPTS: -x -e SC2059 -e SC2086
+        with:
+          scandir: '.github/workflows/scripts'
+          severity: warning
```

Comment:
Note: If we don't pin the version of `shellcheck` tool to a specific version, we risk getting errors after shellcheck is upgraded. From my experience, shellcheck tends to introduce changes in new versions that cause the tool to fail on scripts that it would previously succeed on.